apis/projectcontour/v1/httpproxy.go

```diff
@@ -58,36 +58,36 @@ type Condition struct {
 	Header *HeaderCondition `json:"header,omitempty"`
 }
 
-// HeaderCondition specifies the header condition to match.
-// Name is required. Only one of Present or Contains must
-// be provided.
+// HeaderCondition specifies how to conditionally match against HTTP
+// headers. The Name field is required, but only one of the remaining
+// fields should be be provided.
 type HeaderCondition struct {
-
-	// Name is the name of the header to match on. Name is required.
+	// Name is the name of the header to match against. Name is required.
 	// Header names are case insensitive.
 	Name string `json:"name"`
 
-	// Present is true if the Header is present in the request.
+	// Present specifies that condition is true when the named header
+	// is present, regardless of its value. Note that setting Present
+	// to false does not make the false true if the named header is absent.
```

Comment:
```suggestion
	// to false does not make the Headercondition true if the named header is absent.
```
The "false true" part is a little tricky to parse, is this correct?

Reply:
Good catch, fixed.